torch/distributed/_tensor/api.py

```diff
@@ -649,6 +650,69 @@ def distribute_tensor(
         # 1. if the we can further shard this DTensor if the two device mesh belong to
         #   the same parenet mesh and further sharding is possible.
         # 2. check if device mesh and placements are the same
+
+        # use case: right-to-left tensor distribution i.e. nested distribute_tensor()
+        # over 1-d meshes for composable parallelisms. This branch specifically allows
+        # for FSDP2 sharding where the tensor is first sharded on a 1-d device mesh and
+        # then passed to another distribute_tensor() call along with another 1-d device
+        # mesh, and potentially so on.
+        input_parent_mesh = _mesh_resources.get_parent_mesh(device_mesh)
+        dtensor_parent_mesh = _mesh_resources.get_parent_mesh(tensor.device_mesh)
+        right_to_left_distribute = (
+            device_mesh.ndim == 1  # the device mesh arg must be a 1-d mesh
+            and len(placements) == 1  # must be True for 1-d mesh
+            and input_parent_mesh is not None
+            and input_parent_mesh == dtensor_parent_mesh  # distribute over the same mesh
+        )
```

Comment:
Can we avoid using another indention and instead move the below return tensor and raise exception here? Then we don't need another indention.